tests/internals/cache_metrics/cache_metrics_test.go

```diff
@@ -0,0 +1,212 @@
+	KubectlApplyWithTemplate(t, data, "monitoredDeploymentTemplate", monitoredDeploymentTemplate)
+
+	// initial replica count for a deployment
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, 1, 60, 1),
+		"replica count should be 1 after 1 minute")
+
+	// initial replica count for a monitored deployment
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, monitoredDeploymentName, testNamespace, defaultMonitoredDeploymentReplicas, 60, 1),
+		fmt.Sprintf("replica count should be %d after 1 minute", defaultMonitoredDeploymentReplicas))
+
+	data.EnableUseCachedMetrics = true
+	KubectlApplyWithTemplate(t, data, "scaledObjectTemplate", scaledObjectTemplate)
+
+	// Metric Value = 8, DesiredAverageMetricValue = 2
+	// should scale in to 8/2 = 4 replicas, irrespective of current replicas
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, 4, 60, 1),
+		"replica count should be 4 after 1 minute")
+
+	// Changing Metric Value to 4, but because we have a long polling interval, the replicas number should remain the same
+	data.MonitoredDeploymentReplicas = 4
+	KubectlApplyWithTemplate(t, data, "monitoredDeploymentTemplate", monitoredDeploymentTemplate)
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, monitoredDeploymentName, testNamespace, 4, 60, 1),
+		fmt.Sprintf("replica count should be %d after 1 minute", defaultMonitoredDeploymentReplicas))
+
+	// Let's wait at least 60s
+	// the usual setting for `horizontal-pod-autoscaler-sync-period` is 15s)
+	AssertReplicaCountNotChangeDuringTimePeriod(t, kc, deploymentName, testNamespace, 4, 60)
+
+	KubectlDeleteWithTemplate(t, data, "scaledObjectTemplate", scaledObjectTemplate)
+}
+
+func testDirectQuery(t *testing.T, kc *kubernetes.Clientset, data templateData) {
+	t.Log("--- testing query metrics directly ---")
+
+	KubectlApplyWithTemplate(t, data, "deploymentTemplate", deploymentTemplate)
+	data.MonitoredDeploymentReplicas = defaultMonitoredDeploymentReplicas
+	KubectlApplyWithTemplate(t, data, "monitoredDeploymentTemplate", monitoredDeploymentTemplate)
+
+	// initial replica count for a deployment
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, 1, 60, 1),
+		"replica count should be 1 after 1 minute")
+
+	// initial replica count for a monitored deployment
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, monitoredDeploymentName, testNamespace, defaultMonitoredDeploymentReplicas, 60, 1),
+		fmt.Sprintf("replica count should be %d after 1 minute", defaultMonitoredDeploymentReplicas))
+
+	data.EnableUseCachedMetrics = false
+	KubectlApplyWithTemplate(t, data, "scaledObjectTemplate", scaledObjectTemplate)
+
+	// Metric Value = 8, DesiredAverageMetricValue = 2
+	// should scale in to 8/2 = 4 replicas, irrespective of current replicas
+	assert.True(t, WaitForDeploymentReplicaReadyCount(t, kc, deploymentName, testNamespace, 4, 60, 1),
```

Comment:
I'd give more time, something like 2 minutes or 3, this helper will finish once the target replica count is reached, and 1 minute having to scale to 4, could not be enough time

Reply:
@zroubalik Is this resolved?